src/main/java/com/github/manolo8/darkbot/core/manager/HeroManager.java

```diff
@@ -296,7 +291,13 @@ public boolean setMode(@NotNull ShipMode mode) {
     private void setConfigAndFormation(ShipMode mode) {
         if (mode.getConfiguration() != null &&
                 mode.getConfiguration() != getConfiguration()) toggleConfiguration();
-        setFormation(mode.getFormation());
+
+        SelectableItem.Formation formation = mode.getFormation();
+
+        if (formation == null && mode instanceof MutableShipMode) //todo remove me later
```

Comment:
both MutableShipMode, or Config.ShipMode, both have a formation to set the legacy way

also, the MutableShipMode should only do the legacy way *if* the set ship mode that it was set to is a legacy one, if it was a modern ship mode it should be null and this codepath should not be used